name: Pretter

on:
  pull_request:
    branches:
      - main

env:
  FILE_PATTERN: "{out,__tests__}/**/*.{html,css,js,jsx}"

jobs:
  lint:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v4
      - uses: actions/setup-node@v4
        with:
          node-version: 20.x
      - name: Run Prettier
        run: |
          npm install
          npm run prettier --write ./out || echo "::error::Tests failed"
          npm run markuplint || echo "::error::Tests failed"

      - name: Get error log
        id: error_log
        run: echo "Error log here"

      - name: Comment on pull request
        if: ${{ failure() }}
        uses: actions/github-script@v5
        with:
          github-token: ${{ secrets.GITHUB_TOKEN }}
          script: |
            const errorLog = '${{ steps.error_log.outputs.stdout }}';
            const context = github.context;
            const octokit = github.getOctokit(process.env.GITHUB_TOKEN);

            octokit.issues.createComment({
              owner: context.repo.owner,
              repo: context.repo.repo,
              issue_number: context.issue.number,
              body: 'CI failed due to the following error:\n\n```\n' + errorLog + '\n```'
            });
